### PR TITLE
Migrate general code quality checks to github actions

### PR DESCRIPTION
**Pull request by rvolosatovs**
#### Summary

References #2280 

#### Changes

- Migrate general code quality checks to github actions

#### Testing

In a fork

##### Regressions

None

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md`. The target branch is set to `master` if the changes are fully compatible with existing API, database, configuration and CLI.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by rvolosatovs** [comment]
This workflow should also be set as "required" for merge

File: .github/workflows/general.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  quality:
+    name: Code Quality
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Set up Node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: '14'
+    - name: Set up Go
+      uses: actions/setup-go@v2
```
Comment (htdvisser) [inline .github/workflows/general.yml:15]: Let's use

```yml
with:
  go-version: '~1.14'
```

File: .github/workflows/general.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+jobs:
+  quality:
+    name: Code Quality
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Set up Node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: '14'
```
Comment (htdvisser) [inline .github/workflows/general.yml:13]: Shouldn't this be `~14`?

File: .github/workflows/general.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    - name: Set up Go
+      uses: actions/setup-go@v2
+    - name: Initialize Go cache
```
Comment (htdvisser) [inline .github/workflows/general.yml:16]: Same comments as in my other reviews

File: .github/workflows/general.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          ${{ runner.os }}-go-
+    - name: Check out code
+      uses: actions/checkout@v2
+    - name: Editor config
+      uses: snow-actions/eclint@v1.0.1
+      # TODO: Fix EditorConfig errors and remove
+      # https://github.com/TheThingsNetwork/lorawan-stack/issues/2723
+      continue-on-error: true
+    - name: Make Mage
+      run: make tools/bin/mage
```
Comment (htdvisser) [inline .github/workflows/general.yml:31]: Same comments as in my other reviews

**Pull request merged**